### ISSUE TITLE
Formatter cannot be customized

### ISSUE THREAD

**Issue by dereli** [redhat-developer/vscode-java#798]
Formatter cannot be customized. I tried to download the XML file, put it on disk, and point to it in the settings with the correct profile name. I also tried to set formatting options through `.settings/org.eclipse.jdt.core.prefs` file. Same result. I am much probably missing something but cannot figure it out. Is there a way to dump the formatter options after they're applied?

##### Environment
- Operating System: MacOS Mojave 10.14.3 (18D42)
- JDK version: 1.8.0_192-b12
- Visual Studio Code version: 1.31.1 (1.31.1)
- Java extension version: Version 1.31.1 (1.31.1)

##### Steps To Reproduce
1. Open a Java file. Format it.
2. Customize formatting options (either via xml file or via .prefs file)
3. Go back to file and reformat it.
4.  If I manually apply formatting based on the new settings. The old formatting is automatically applied.

```
[Trace - 8:47:49 AM] Sending request 'textDocument/formatting - (1118)'.
Params: {
    "textDocument": {
        "uri": "file:///Users/.......java"
    },
    "options": {
        "tabSize": 4,
        "insertSpaces": true
    }
}


[Trace - 8:47:49 AM] Received response 'textDocument/formatting - (1118)' in 27ms.
Result: [
    {
        "range": {
            "start": {
                "line": 38,
                "character": 45
            },
            "end": {
                "line": 38,
                "character": 45
            }
        },
        "newText": " "
    },
    ....
]
```
##### Current Result
No new formatting is applied.

##### Expected Result
Expecting format changes expected based on the customizations. 

##### Additional Informations
The extension just works fine autoformat on save, auto organize imports etc.

**Comment by fbricon** [comment]
Are you trying to format a standalone java file or a file inside a Java project?

**Comment by fbricon** [comment]
which particular formatting settings do you not see being applied?

**Comment by snjeza** [comment]
@dereli you may want to take a look at https://github.com/redhat-developer/vscode-java/wiki/Formatter-settings and https://github.com/eclipse/eclipse.jdt.ls/pull/640

**Comment by dereli** [comment]
@fbricon I always tried to format a file as a part of a project. Now I tried standalone formatting, it works as expected. That probably means, a setting somewhere in the project is overriding or blocking the formatting. Any possible culprits?

The settings I tried is below. Even though I don't think any individual settings is the issue. (btw inconsistency in spacing is for testing purposes).

```
<setting id="org.eclipse.jdt.core.formatter.insert_space_after_opening_brace_in_array_initializer" value="do not insert" />
<setting id="org.eclipse.jdt.core.formatter.insert_space_before_closing_brace_in_array_initializer" value="insert" />

```
